kernel/kernel-api/src/test/scala/io/delta/kernel/internal/actions/ProtocolSuite.scala

```diff
@@ -369,4 +374,24 @@ class ProtocolSuite extends AnyFunSuite {
         assert(merged.getWriterFeatures.asScala === expWriterFeatures)
       }
   })
+
+  test("extract protocol from the row representation") {
+    val values: util.Map[Integer, Object] = {
+      val map = new util.HashMap[Integer, Object]()
```

Comment:
nit: this map = { create a map, then return a map } seems a bit odd? could you just (a) do an inline map in scala and then use `.asJava` or (b) just create and mutate the `values` map itself (i.e. no inner `map`?)